conclave-host/src/main/kotlin/com/r3/conclave/host/internal/gramine/GramineEnclaveHandle.kt

```diff
@@ -52,8 +59,27 @@ class GramineEnclaveHandle(
 
         /** Create a socket host interface. */
         enclaveInterface = SocketHostEnclaveInterface()
+
+        dockerImageTag = retrieveImageTagFromManifest()
     }
 
+    private fun String.mapWorkingDirectory(): String {
+        return this.replace(workingDirectory.toFile().absolutePath, DOCKER_WORKING_DIR).replace("\\", "/")
+    }
+
+    private fun runSimpleCommand(command: List<String>): String {
+        val process = ProcessBuilder()
+            .command(command)
+            .start()
+        process.waitFor(5L, TimeUnit.SECONDS)
+        val lines = process.inputStream.bufferedReader().use(BufferedReader::readText).trimEnd()
+            .split(System.lineSeparator())
+        //  We expect 1 line result and a newline
+        check(lines.size == 1) { "Command produced an unexpected result" }
```

Comment:
If this is only needed to get the `id` then just return the whole string. no need to parse it into lines and then assert there's only one.

Reply:
Ok, just wanted to prevent potential failures. I noticed that we used `LinuxExec.exec` without checking the (null) return value. This has issues when I used Docker, i.e. failures in running commands inside Docker weren't propagated and no exceptions were thrown